components/search-modal/element.js

```diff
@@ -60,9 +62,15 @@ export class MDNSearchModal extends L10nMixin(LitElement) {
   }
 
   /** @param {InputEvent} event */
-  _input({ target }) {
+  _input({ inputType, target }) {
     if (target instanceof HTMLInputElement) {
       this._query = target.value;
+      if (!this._hasChanged && inputType.startsWith("insert")) {
+        this._hasChanged = true;
+        gleanClick(
+          `quick-search-change: ${inputType === "insertFromPaste" ? "paste" : "type"}`,
```

Comment:
This seems unnecessary information to capture: let's say we find the majority of users e.g. paste inputs rather than type them in, what would we do with that information?

Reply:
The main point here is that we're capturing search engagement, and while we do this, we can distinguish typing from pasting for free.